lib/benchee/benchmark/collect/return_value.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Benchee.Benchmark.Collect.ReturnValue do
+  @moduledoc false
+
+  # Returns the value of the evaluated function.
+  # Used for pre checks.
+
+  # Does not strictly speaking implement Benchee.Benchmark.Collect because
+  # it does not return a metric.
+
+  @spec collect((-> any)) :: {any, any}
+  def collect(function) do
+    result = function |> Task.async() |> Task.await(:infinity)
```

Comment:
This is curios to me - why did you decide to implement it with `Task.async` & `await`? As we're immediately awaiting there is no time benefit - copying the function including closure to the process likely makes it slower.

I think I can see an argument for isolating the different functions & inputs being run - but then I think I'd rather do it "on the outside" - i.e. wrapping the invocation of this inside the `pre_check` function.

Reply:
I think it was for isolation indeed since it seemed other collectors were all running it in a new process and I tried to make it consistent - I just noticed it isn't the case for `Benchee.Benchmark.Collect.Time` so there is no reason indeed 😁

Should be:
```suggestion
    result = function.()
```

Reply:
yeah reductions and memory run in a separate process to isolate their respective measurements :)